### ISSUE TITLE
BUG,DOC,BLD: parse {@param, @type, @returns, @rtype} docstrings

### ISSUE THREAD

**Issue by westurner** [RDFLib/rdflib#569]
https://rdflib.readthedocs.org/en/latest/apidocs/rdflib.plugins.parsers.html#rdflib.plugins.parsers.structureddata.RDFaParser.parse

Epydoc
- | Docs: http://epydoc.sourceforge.net/fields.html

sphinx-epytext
- | Pypi: https://pypi.python.org/pypi/sphinx-epytext

sphinxcontrib-napoleon
- | Docs: https://pypi.python.org/pypi/sphinxcontrib-napoleon 
- [x] conf.py: extensions.append('sphinx_epytext')
  - numpy & google, khan style: extensions.append(sphinxcontrib.napoleon OR sphinx.ext.napoleon)
  - [x] epydoc style docstrings: extensions.append('sphinx_epytext')
- [x] docs/sphinx-requirements.txt: 'sphinx-epytext'

**Comment by westurner** [comment]
See PR #570

**Comment by mwatts15** [comment]
@westurner could you explain in plain English what problem you are identifying here and fixing in #570?

**Comment by westurner** [comment]
When you read the API docs, the argument and return type specifications
present in the function and method docstrings are omitted.
#570 adds a Sphinx extension so that docstrings are parsed and transformed

For example, these docstrings contain structured docstrings data elements
that the rdflib docs omit due to lack of this additional configuration:
https://github.com/RDFLib/pyrdfa3/blob/master/pyRdfa/parse.py

https://rdflib.readthedocs.io/en/latest/apidocs/rdflib.plugins.parsers.pyRdfa.html#parse-module

... It's sort of ironic because these docstring attributes and values are
almost RDF, which rdflib is all about; but there's as of yet no way (other
than direct inclusion of RDFa) to include RDFa in sphinx-generated HTML
with Sphinx and the docutils doctree.

On Saturday, October 12, 2019, Mark Watts <notifications@github.com> wrote:
> @westurner could you explain in plain English what problem you are
identifying here and fixing in #570?

**Comment by mwatts15** [comment]
Thanks. It looks like pyrdfa modules were removed in `master` last year and the the current `master` has only a few uses of epydoc syntax for method documentation in operators.py. Probably that just needs to be changed to match the rest of the docs.
